sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -659,604 +600,482 @@ protected EventProcessorClient()
+                await _partitionClosingAsync(eventArgs).ConfigureAwait(false);
             }
 
-            var processingTask = RunPartitionProcessingAsync(partitionId, startingPosition, RunningTaskTokenSource.Token);
-
-            var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(RunningTaskTokenSource.Token);
-            ActivePartitionProcessors[partitionId] = (processingTask, tokenSource);
-            Logger.StartPartitionProcessingComplete(Identifier);
+            PartitionStartingPositionDefaults.Remove(partition.PartitionId);
         }
 
         /// <summary>
-        ///   Stops an owned partition processing task in case it is running.  It is also removed from the tasks dictionary
-        ///   along with its corresponding token source.
+        ///   Creates a <see cref="StorageManager" /> to use for interacting with durable storage.
         /// </summary>
         ///
-        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is being stopped.</param>
-        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
-        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <param name="checkpointStore">The client responsible for interaction with durable storage, responsible for persisting checkpoints and load-balancing state.</param>
         ///
-        private async Task StopPartitionProcessingIfRunningAsync(string partitionId,
-                                                                 ProcessingStoppedReason reason,
-                                                                 CancellationToken cancellationToken)
-        {
-            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
-            Logger.StopPartitionProcessingStart(partitionId, reason);
+        /// <returns>A <see cref="StorageManager" /> with the requested configuration.</returns>
+        ///
+        private StorageManager CreateStorageManager(BlobContainerClient checkpointStore) => new BlobsCheckpointStore(checkpointStore, RetryPolicy);
 
-            if (ActivePartitionProcessors.TryRemove(partitionId, out var activeTaskAndTokenSource))
-            {
-                var (processingTask, tokenSource) = activeTaskAndTokenSource;
+        /// <summary>
+        ///   Processes the starting positions for checkpoints, ensuring that any overrides set by the <see cref="PartitionInitializingAsync" />
+        ///   handler are respected.
+        /// </summary>
+        ///
+        /// <param name="sourceCheckpoints">The checkpoint set to process.</param>
+        ///
+        /// <returns>An enumerable of the <paramref name="sourceCheckpoints"/> with any overrides applied to the starting position.</returns>
+        ///
+        /// <remarks>
```

Comment:
I don't understand what this remark is saying.  We don't mutate `sourceCheckpoints` at all (and we can't really, since it's `IEnumerable<T>`  We also never mutate any of the items inside the enumeration here.  Is this a hold over from an earlier design?

Reply:
Yeah, that does reflect an earlier design.   I'll tune it up.   

_(we had originally created an artificial checkpoint for every partition with the starting location and then overwrote that with the natural checkpoint data from storage.)_